extensions/functions_comparison.yaml

```diff
@@ -62,6 +62,9 @@ scalar_functions:
     description: Whether a value is null.
     impls:
       - args:
+          - name: nan_is_null
```

Comment:
Would you mind elaborating on the motivation for adding this option? I'm wondering why an `is_nan` function wouldn't suffice.

Reply:
Different systems have different definitions of whether NaN is equivalent to NULL or not.  

This is something we encountered when working on the Acero compute functions; https://issues.apache.org/jira/browse/ARROW-12959 and https://issues.apache.org/jira/browse/ARROW-12960.

If we wanted `is_null()` to return `FALSE` for `NaN` values and `TRUE` for `NULL`, then I don't think mapping it to `is_nan()` would achieve this, as that would return `FALSE` for `NULL`.  Or would it return `NULL` for `NULL`? Either way, I guess having this explicitly as an option removes that ambiguity.

Reply:
Is there a reason producers can't handle this kind of transform?

Most systems treat nan and null as distinct. As far as I know, it's only R and Pandas that support treating these two values equivalently in some contexts but not all.

Ideally we can avoid propagating the mistake of treating these two values as optionally equivalent as part of the spec.

Reply:
+1 for @cpcloud. I don't see why these shenanigans couldn't be modeled without having this option. Let's not embed arbitrary and questionable special cases into Substrait core.